ui/src/store/features/github/async-thunk/fetch-user-organizations.ts

```diff
@@ -0,0 +1,38 @@
+import { ComboboxItem } from '@/components';
+import { RootState } from '@/store';
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { GithubClient, UserData } from '../github-client';
+import { githubActions } from '../github-slice';
+
+export const fetchUserAndOrgsThunk = createAsyncThunk(
+  'github/fetchUserAndOrgs',
+  async (_, { dispatch, getState }) => {
+    const { token, queryUserAndOrganizations } = (getState() as RootState)
+      .github;
+
+    if (queryUserAndOrganizations === 'loading') return;
+
+    try {
+      dispatch(githubActions.setQueryUserState('loading'));
+
+      const githubClient = new GithubClient(token);
+      const user = await githubClient.fetchUser();
+      const orgs = await githubClient.fetchOrgs();

```

Comment:
For situations where the requests are complementary and not dependent on each other we can use `Promise.all` to trigger them concurrently and improve the response time.
```suggestion
const [user, orgs] = await Promise.all([githubClient.fetchUser(), githubClient.fetchOrgs()]);
```